crates/ra_project_model/src/lib.rs

```diff
@@ -24,6 +25,34 @@ pub use crate::{
     sysroot::Sysroot,
 };
 
+#[derive(Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum InlayHintDisplayType {
```

Comment:
To my mind, the best place for this type would be at `crates/ra_ide/src/inlay_hints.rs` itself. So that `crates/rust-analyzer` imports them into it's `Config` struct.
If I am not mistaken `ra_ide` doesn't depend on `serde`, so we could make a newtype in `crates/rust-analyzer` and declare it deserializable there. Do I have a point, @SomeoneToIgnore ?

Reply:
Yes, that's totally correct, adding them to an extra dependency crate seems not to be worth it.
Here's the discussion ticket, for the context: https://github.com/rust-analyzer/rust-analyzer/issues/3189

Reply:
Yeah, this should be a POD defined in `ra_ide` crate. It should **not** implement serialize/deserialize, as ra_ide should be abstracted away from any external repsresnetation. Serialization/deserliazation should be done in `rust-analyzer` crate, even if means duplicating the type definition.

Reply:
If I move it to the ra_ide crate how do I mark it as deserializable in the rust-analyzer crate? If I duplicate the type definitions then the types won't be the same?

Reply:
@slyngbaek we may use `remote` attribute as per [serde docs](https://serde.rs/remote-derive.html), though yes we will need to copy the struct definition (due to the unavoidable limitation of procedural macros)

Reply:
Gotcha, didn't know you could do that. Unfortunately, there's still limited support for the `with` attribute. https://github.com/serde-rs/serde/issues/1005 eg. it doesn't work with containers like we're doing with `Vec<InlayKind>`

Reply:
Actually, I found a work-around until better support is added in serde. https://github.com/serde-rs/serde/issues/723#issuecomment-382501277

Reply:
Hmm, I myself didn't use that `serde` feature. I hope @matklad has the better solution here.
Because what comes to my mind is:
 if you say it won't work, I'd bail to just copying the definition and making a mapping:
```rust
// rust-analyzer/config.rs
struct InlayHintOpts { /* copy of ra_ide::InlayHintsOpts definition */ }
impl From<ra_ide::InlayHintsOpts> for InlayHintsOpts { /* I wish there was a macro for this */ }
impl From<InlayHintsOpts> for ra_ide::InlayHintsOpts { /* I wish there was a macro for this */ }
```